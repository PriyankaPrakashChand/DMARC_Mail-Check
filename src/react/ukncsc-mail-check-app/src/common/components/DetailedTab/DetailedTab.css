.DetailedTab {
  display: block !important;
  color: black !important;
  text-align: left !important;
  padding: 0px !important;
}

.DetailedTab--content {
  background: #dee0e2 !important;
  margin-right: 5px !important;
  margin-left: 5px !important;
  margin-bottom: 10px !important;
  padding-top: 10px !important;
  padding-bottom: 10px !important;
}

.DetailedTab--content > p {
  margin-bottom: 0px !important;
  padding-left: 10px !important;
}

.DetailedTab--content > h1 {
  padding-left: 10px !important;
  margin-top: 0px !important;
  margin-bottom: 0px !important;
}

.DetailedTab--content > h5 {
  padding-left: 10px !important;
  margin-top: 0px !important;
}

.DetailedTab.active {
  border-top: 5px solid #34363c !important;
  margin-top: -5px;
  border-left: 1px solid #bfc1c3 !important;
  border-right: 1px solid #bfc1c3 !important;
}

.DetailedTab.active > .DetailedTab--content {
  background: #ffffff !important;
}
