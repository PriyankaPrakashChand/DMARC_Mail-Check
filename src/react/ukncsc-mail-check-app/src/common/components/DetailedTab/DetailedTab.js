import React, { Component } from 'react';
import { Menu } from 'semantic-ui-react';

import './DetailedTab.css';

class DetailedTab extends Component {
  componentWillMount() {
    const { children, index, tabSelected } = this.props;
    if (index === 0) {
      tabSelected(children, index);
    }
  }

  render() {
    const {
      title,
      headline,
      subtitle,
      index,
      activeIndex,
      tabSelected,
      children,
    } = this.props;
    return (
      <Menu.Item
        active={index === activeIndex}
        onClick={() => tabSelected(children, index)}
        className="DetailedTab"
      >
        <div className="DetailedTab--content">
          <p>{title}</p>
          <h1>{headline}</h1>
          <h5>{subtitle}</h5>
        </div>
      </Menu.Item>
    );
  }
}

export default DetailedTab;
