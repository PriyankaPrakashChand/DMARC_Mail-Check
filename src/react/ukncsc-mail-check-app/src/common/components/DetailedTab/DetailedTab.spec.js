import React from 'react';
import { render } from 'react-testing-library';
import DetailedTab from './DetailedTab';

describe('DetailedTab', () => {
  let container;

  describe('when the tab is the first one', () => {
    let tabSelectedMock;

    beforeEach(() => {
      tabSelectedMock = jest.fn();

      ({ container } = render(
        <DetailedTab
          title="My first tab"
          headline="101"
          subtitle="cool"
          index={0}
          activeIndex={0}
          tabSelected={tabSelectedMock}
        >
          <p>Hello World</p>
        </DetailedTab>
      ));
    });

    test('it should match the snapshot', () =>
      expect(container).toMatchSnapshot());

    test('should should ensure that a tab is selected', () =>
      expect(tabSelectedMock).toHaveBeenCalled());
  });
});
