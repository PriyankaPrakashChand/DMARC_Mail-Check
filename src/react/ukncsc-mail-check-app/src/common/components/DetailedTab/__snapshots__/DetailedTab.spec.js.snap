// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedTab when the tab is the first one it should match the snapshot 1`] = `
<div>
  <a
    class="active item DetailedTab"
  >
    <div
      class="DetailedTab--content"
    >
      <p>
        My first tab
      </p>
      <h1>
        101
      </h1>
      <h5>
        cool
      </h5>
    </div>
  </a>
</div>
`;
